be/src/runtime/thread_mem_tracker_mgr.h

```diff
@@ -217,14 +217,14 @@ inline void ThreadMemTrackerMgr::cache_consume(int64_t size) {
     if (_untracked_mem >= config::mem_tracker_consume_min_size_bytes ||
         _untracked_mem <= -config::mem_tracker_consume_min_size_bytes) {
         DCHECK(_untracked_mems.find(_tracker_id) != _untracked_mems.end());
+        start_thread_mem_tracker = false;
```

Comment:
Add comment to explain why adding `start_thread_mem_tracker = false;` here

Reply:
Avoid getting into infinite recursion if there is a temporary memory allocation in mem_tracker.consume/try_consume.

It is mentioned in the comments of `tcmalloc_hook.h`: Allocating memory in the Hook command causes the Hook to be entered again, infinite recursion.

This needs to ensure that all memory allocated in mem_tracker.consume/try_consume is freed in time to avoid tracking misses.

Add comments when modifying uniformly.